src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -126,7 +126,7 @@ public AbstractJob createJob(final CitizenData citizen)
     @Override
     public boolean neededForWorker(@Nullable final ItemStack stack)
     {
-        return Utils.isMiningTool(stack);
+        return Utils.isMiningTool(stack) || neededResources.containsKey(stack.getUnlocalizedName());
```

Comment:
Again this is not the way to store stacks!. If you want to store stacks, either serialize them or use the items registry Name. 
A good example of when this will not work: QED From Extra Utilities ist Unlocalized Name changes!